rest-api/dao/biobank_order_dao.py

```diff
@@ -265,8 +276,43 @@ def from_client_json(self, resource_json, participant_id=None, client_id=None):
           % (participant_id, resource.subject, self._participant_id_to_subject(participant_id)))
     self._add_identifiers_and_main_id(order, resource)
     self._add_samples(order, resource)
+    if resource.status and not resource.amendedInfo:
+      self._handle_new_status(order, resource)
+    if resource.amendedReason:
+      order.amendedReason = resource.amendedReason
+    if resource.amendedInfo:
+      order.amendedUsername, order.amendedSiteId = self._parse_handling_info(resource.amendedInfo)
+    order.version = expected_version
     return order
 
+  @classmethod
+  def _handle_new_status(cls, order, resource):
+    site_dao = SiteDao()
+    order.status = resource.status
+    if order.status.upper() == 'CANCELLED':
+      order.orderStatus = BiobankOrderStatus.CANCELLED
+      site = site_dao.get_by_google_group(resource.cancelledInfo.site.value)
+      if site:
+        order.cancelledSiteId = site.siteId
+      else:
+        raise BadRequest('A valid google_group is required for cancelledInfo.site.value')
+      order.cancelledUsername = resource.cancelledInfo.author.value
+      order.cancelledTime = clock.CLOCK.now()
+      order.amendedReason = resource.amendedReason
+    elif order.status.upper() == 'RESTORED':
```

Comment:
One thing that's a little weird here is we're going to have "RESTORED" in our API coming in for restore operations, resulting in "UNSET" in the response... is that OK?

Reply:
Then this should be UNSET in the request, too, right? The request and response should match...

Reply:
Spoke with HPRO team again, it makes sense to them the way it is now, personally I think sending `unset` in the request would be confusing, if anything I think we should return 'restored' in the payload. I'll leave for now, you and I can discuss further, it will be an easy change whichever we pick.